### PR TITLE
Stats: allow overriding default blog ID via arg in stats_get_csv

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

This would allow sites that use `stats_get_csv` in third-party plugins to fetch stats about different sites they own, all from one site, by specifying a different `blog_id` when calling `stats_get_csv`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

See p1733278660045129-slack-CDD9LQRSN

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Start with 2 sites connected to WordPress.com, where the stats module is active.
2. In a functionality plugin, add the following:

```php
add_action( 'wp_footer', function () {
	$options = array( 'blog_id' => YOUR_BLOG_ID );
	var_dump( stats_get_csv( 'postviews', $options ) );
} );
```

3. View the site's frontend.
    * You should see different stats when changing the blog ID, or removing that custom blog id and leaving `$options` as an empty array.

### CODE REVIEW

File: projects/plugins/jetpack/modules/stats.php [hunk 1]
```diff
@@ -1578,11 +1578,11 @@ function stats_get_csv( $table, $args = null ) {
 		'limit'     => 3,
 		'post_id'   => false,
 		'summarize' => '',
+		'blog_id'   => Jetpack_Options::get_option( 'id' ),
```
Comment (raamdev) [inline projects/plugins/jetpack/modules/stats.php:1582]: We should provide a PHPDoc type hint for the `blog_id` argument in the Doc block.

**Review by raamdev — DISMISSED** [review]
This looks good to me. I made one suggestion to document the new argument in the PHPDoc block.

File: projects/plugins/jetpack/modules/stats.php (hunk 1 above)
Comment (jeherve) [reply projects/plugins/jetpack/modules/stats.php:1582]: That's a good point. Done in 43d3c7c9de3cf9a10138e12effec4a8b9e58050f

**Review by raamdev — APPROVED** [review]
This looks great to me. ✅

**Pull request merged**